securitas/utility/messaging.py

```diff
@@ -0,0 +1,26 @@
+import sys
+import traceback
+
+from fedora_messaging import api, exceptions as fml_exceptions
+import backoff
+
+from securitas import app
+
+def backoff_hdlr(details):
+    app.logger.error(f"Publishing message failed. Retrying. {traceback.format_tb(sys.exc_info()[2])}")
+
+@backoff.on_exception(
+    backoff.expo,
+    (fml_exceptions.ConnectionException, fml_exceptions.PublishReturned),
+    max_tries=3,
+    on_backoff=backoff_hdlr)
+def _publish(message):
+    api.publish(message)
+
+def publish(message):
+    try:
+        _publish(message)
+    except (fml_exceptions.ConnectionException, fml_exceptions.PublishReturned) as e:
```

Comment:
Here I think you can simply catch `fml_exceptions.BaseException`, because you want the message to be logged whether it's a temporary error or not.